digits/static/js/layer_visualizations/readCaffeModel.js

```diff
@@ -0,0 +1,163 @@
+function getTreeData(type,text){
+
+  var fileReader, lines;
+  window.couplings = new Array();
+
+  // *** Prototxt Formatting helpers ****
+
+  // Cleans up attributes loaded from parsing a prototxt file
+  function cleanAttr(attr,name){return attr.replace(name,'').replace(/["" '']/g,'');}
+
+
+  // Get attributes from an string blob loaded from a prototxt file
+  function getAttr(data,name){
+    var attr = _.filter(data,function(d){return d.indexOf(name) > -1});
+    if (attr.length > 1 && name == "bottom:"){addCouples(data,attr);}
+    var d = attr[0];
+    if (_.isUndefined(d)){return undefined}
+    return cleanAttr(d,name);
+  }
+
+  // Get the location of all layer objects in prototxt file
+  function getLayerIndeces(data){
+    return _.without(_.map(data,function(d,i){
+      return d.indexOf("layer {") > -1 || d.indexOf("layers {") > -1 ? i : null;
+    }),null);
+  }
+
+  // Although only sort tree via one "top", keep track of all tops incase
+  // they are referenced as the parent for a future non-child node
+  function getTops(data){
+    var tops = _.filter(data,function(d){return d.indexOf("top:") > -1});
+    return _.map(tops, function(top){return cleanAttr(top,"top:")});
+  }
+
+  // If a node has multiple bottoms / inputs, add them to a separate array
+  function addCouples(data,bottoms){
+    var top  = getAttr(data,"top:");
+    var name = getAttr(data,"name:");
+    _.each(bottoms, function(b){
+      var bottom = cleanAttr(b,"bottom:");
+      couplings.push({parent: bottom, child: top, name: name});
+    });
+  }
+
+  // *** Read Prototxt File ****
+  if (type == 'file'){
+    fileReader = new XMLHttpRequest();
+    fileReader.open("GET", text, false);
+
+    fileReader.onreadystatechange = function (){
+      if(fileReader.readyState === 4){
+        if(fileReader.status === 200 || fileReader.readyState == 4){
```

Comment:
In line 50 you say
`if(fileReader.readyState === 4){`
and immediatelly after
`if(fileReader.status === 200 || fileReader.readyState == 4){`
So the latter is always true if the former is true. I bet you meant to say 
`(fileReader.status === 200 || fileReader.status == 0)` here.